### ISSUE TITLE
Hooks: userland options are not merged with defaults

### ISSUE THREAD

**Issue by vxna** [thebuilder/react-intersection-observer#141]
I am neither familiar with `flow` nor I have a valid fix, but I assume it happens since this [declaration](https://github.com/thebuilder/react-intersection-observer/blob/master/src/index.js#L163-L165)
```js
export function useInView(
  ref: React.ElementRef<*>,
  options: IntersectionOptions = {
    threshold: 0,
  },
)
```

compiles into
```js
function useInView(ref, options) {
  if (options === void 0) {
    options = {
      threshold: 0
    };
  }
```

Repro: https://codesandbox.io/s/42l85v1vv0

**Comment by thebuilder** [comment]
Nice catch - this actually turned out to be a more servere bug inside `intersection.js`.
It was suppose to also ensure a threshold of at least 0, which it did when observering. But this wasn't kept in the options object that it checked against inside the `onChange` method.

I'll publish a fix, that will also remove `threshold: 0` from the default options, and instead ensure the threshold is defined when the observer is created.

**Comment by thebuilder** [comment]
Fixed in v6.3.1! 🎉
